streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -590,18 +591,18 @@ public void process(final ProcessorSupplier<? super K, ? super V> processorSuppl
     /**
      * Repartition a stream. This is required on join operations occurring after
      * an operation that changes the key, i.e, selectKey, map(..), flatMap(..).
-     *
-     * @param joined joined control object
-     * @return a new {@link KStreamImpl}
      */
-    private KStreamImpl<K, V> repartitionForJoin(final Joined<K, V, ?> joined) {
-        final Serde<K> repartitionKeySerde = joined.keySerde() != null ? joined.keySerde() : keySerde;
-        final Serde<V> repartitionValueSerde = joined.valueSerde() != null ? joined.valueSerde() : valSerde;
+    private KStreamImpl<K, V> repartitionForJoin(final String repartitionNameOverride,
+                                                 final Serde<K> keySerdeOverride,
+                                                 final Serde<V> valueSerdeOverride) {
+        final String name = repartitionNameOverride != null ? repartitionNameOverride : this.name;
```

Comment:
nit:  `repartitionNameOverride` won't ever be `null` from doing the checks for the name on lines 571 and 576, so we can get rid of this line

Reply:
Thanks for the feedback, but I think it would be null in the call from https://github.com/apache/kafka/pull/5741/files#diff-5142e1d4a6410459d6bf6df98828e5afL682 and https://github.com/apache/kafka/pull/5741/files#diff-5142e1d4a6410459d6bf6df98828e5afL703

Reply:
Yes that's correct as this PR stands now.   
But if we put the name check back to what it was originally, then this line is not needed.

Reply:
Ok, I've pushed the check back out to the callers.